### PR TITLE
feat(filter): loose tag regex and compatibility

### PR DESCRIPTION
**Pull request by Tetrergeru**
by @lordvidex

### CODE REVIEW

File: cmd/filter/compatibility.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/moira-alert/moira/filter"
+)
+
+type Compatibility struct {
```
Comment (kissken) [inline cmd/filter/compatibility.go:9]: missed godoc

File: filter/compatibility.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+const (
+	StrictStartMatch RegexTreatment = false
+	LooseStartMatch  RegexTreatment = true
+)
+
+type SingleStarMatch bool
+
+const (
+	SingleStarMatchDisabled    SingleStarMatch = false
```
Comment (kissken) [inline filter/compatibility.go:18]: maybe private?

and godoc

File: filter/series_by_tag_pattern_index_test.go [hunk 3]
```diff
@@ -172,25 +179,39 @@ func TestSeriesByTagPatternIndex(t *testing.T) {
...
 				{"name", MatchOperator, "cpu"},
-				{"tag1", EqualOperator, "val1"}},
+				{"tag1", EqualOperator, "val1"},
+			},
+			"name=~test1": {
+				{"name", MatchOperator, "test1"},
+			},
+			"tag1=~al1": {
+				{"tag1", MatchOperator, "al1"},
+			},
```
Comment (kissken) [inline filter/series_by_tag_pattern_index_test.go:207]: вот тут можем докинуть тест
на такое еще
tag2=~.*

File: filter/compatibility.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package filter
+
+type Compatibility struct {
```
Comment (kissken) [inline filter/compatibility.go:3]: и мб запривалить от греха

File: filter/compatibility.go (hunk 2 above)
Comment (Tetrergeru) [reply filter/compatibility.go:18]: It is used in filter, when feature flags are parsed from string into these enums

**Review by kissken — APPROVED** [review]

**Review by msaf1980 — APPROVED** [review]

**Review by almostinf — APPROVED** [review]

**Pull request merged**